python/az/aro/azext_aro/custom.py

```diff
@@ -271,9 +271,12 @@ def get_route_tables_from_subnets(cli_ctx, subnets):
     for sn in subnets:
         sid = parse_resource_id(sn)
 
-        subnet = network_client.subnets.get(resource_group_name=sid['resource_group'],
-                                            virtual_network_name=sid['name'],
-                                            subnet_name=sid['resource_name'])
+        if not all(key in sid.keys() for key in ('resource_group','name','resource_name')):
+            raise ResourceNotFoundError("subnet is empty")
```

Comment:
I think we need to be more descriptive here.  `subnet is empty` will confuse end users running these `az aro` commands receiving the output. Additionally, we can set recommendations in errors we raise which I think we should be doing in this specific instance.

The reasoning behind more specificity would be along the lines of customers questioning the issue with:
* What does "subnet is empty" mean? 
* Why is the subnet empty?  
* How do I fix this issue?

Reply:
Adding a bunch of context here for other reviewers regarding enrichment (which I think may be helpful for review of this)

We have an enricher factory which runs on the RP frontend process.  This runs for 10 seconds max during any create/read/update/delete operation, and queries the cluster's kube-api server to attempt to enrich the cluster doc with worker profiles, cluster version information, cluster service principal, and ingress profiles.  It is a best-effort enricher and if it fails the frontend process will continue on normally.

For our write operations in the `az aro` command module, we attempt to fetch the cluster first, which will invoke the enricher on the frontend for 10 seconds to attempt to fetch this information from the cluster directly by talking to the kube-apiserver. If it fails, it will not affect any other frontend operations from continuing, but may result in failures later down the pipeline (such as one described in the situation in this user story)

If the enricher fails to populate any worker profiles, the python code cannot ensure the required permissions on route tables and other network resources, resulting in a failure from our side and poor customer experience on a python error. 

@SrinivasAtmakuri I think we should do the following in this situation:
1. Update the error message within the `ResourceNotFoundError` to something along the lines of `Failed to validate subnet '_subetName_' 
2. Provide a recommendation of retrying, and if continuous failures, open up an Azure support ticket
3. We might want to rethink the error we're throwing here.  A `ResourceNotFound` doesn't 100% fit the bill.  I'm leaning towards a `ValidationError`, but can be swayed on this.

Reply:
:+1:

Reply:
IIRC we saw this happen when a customer had poorly configured machinesets, so I think that warning and continuing is the better approach, since they couldn't get in to *fix* said machinesets.